lua/nvim-tree/api.lua

```diff
@@ -205,54 +189,59 @@ Api.node.open.tab = wrap_node(open_or_expand_or_dir_up "tabnew")
-Api.marks.bulk.delete = wrap(require("nvim-tree.marks.bulk-delete").bulk_delete)
-Api.marks.bulk.trash = wrap(require("nvim-tree.marks.bulk-trash").bulk_trash)
-Api.marks.bulk.move = wrap(require("nvim-tree.marks.bulk-move").bulk_move)
-Api.marks.navigate.next = wrap(require("nvim-tree.marks.navigation").next)
-Api.marks.navigate.prev = wrap(require("nvim-tree.marks.navigation").prev)
-Api.marks.navigate.select = wrap(require("nvim-tree.marks.navigation").select)
-
-Api.config.mappings.default_on_attach = require("nvim-tree.keymap").default_on_attach
-
-Api.config.mappings.get_keymap = wrap(function()
-  return require("nvim-tree.keymap").get_keymap()
-end)
-
-Api.config.mappings.get_keymap_default = wrap(function()
-  return require("nvim-tree.keymap").get_keymap_default()
-end)
+Api.node.show_info_popup = wrap_node(actions.node.file_popup.toggle_file_info)
+Api.node.run.cmd = wrap_node(actions.node.run_command.run_file_command)
+Api.node.run.system = wrap_node(actions.node.system_open.fn)
+
+Api.node.navigate.sibling.next = wrap_node(actions.moves.sibling.fn "next")
+Api.node.navigate.sibling.prev = wrap_node(actions.moves.sibling.fn "prev")
+Api.node.navigate.sibling.first = wrap_node(actions.moves.sibling.fn "first")
+Api.node.navigate.sibling.last = wrap_node(actions.moves.sibling.fn "last")
+Api.node.navigate.parent = wrap_node(actions.moves.parent.fn(false))
+Api.node.navigate.parent_close = wrap_node(actions.moves.parent.fn(true))
+Api.node.navigate.git.next = wrap_node(actions.moves.item.fn { where = "next", what = "git" })
+Api.node.navigate.git.next_skip_gitignored = wrap_node(actions.moves.item.fn { where = "next", what = "git", skip_gitignored = true })
+Api.node.navigate.git.prev = wrap_node(actions.moves.item.fn { where = "prev", what = "git" })
+Api.node.navigate.git.prev_skip_gitignored = wrap_node(actions.moves.item.fn { where = "prev", what = "git", skip_gitignored = true })
+Api.node.navigate.diagnostics.next = wrap_node(actions.moves.item.fn { where = "next", what = "diag" })
+Api.node.navigate.diagnostics.prev = wrap_node(actions.moves.item.fn { where = "prev", what = "diag" })
+Api.node.navigate.opened.next = wrap_node(actions.moves.item.fn { where = "next", what = "opened" })
+Api.node.navigate.opened.prev = wrap_node(actions.moves.item.fn { where = "prev", what = "opened" })
+
+Api.git.reload = wrap(actions.reloaders.reload_git)
+
+local events = require "nvim-tree.events"
+
+Api.events.subscribe = wrap(events.subscribe)
+Api.events.Event = events.Event
+
+local live_filter = require "nvim-tree.live-filter"
+
+Api.live_filter.start = wrap(live_filter.start_filtering)
+Api.live_filter.clear = wrap(live_filter.clear_filter)
+
+local marks = require "nvim-tree.marks"
+local marks_navigation = require "nvim-tree.marks.navigation"
+local marks_bulk_delete = require "nvim-tree.marks.bulk-delete"
+local marks_bulk_trash = require "nvim-tree.marks.bulk-trash"
+local marks_bulk_move = require "nvim-tree.marks.bulk-move"
```

Comment:
I tried to add the various `init.lua`s in the `marks` module but dependency loops start to come up, so this is best solution I could find.

Reply:
Marks are not quite like the other actions. They need a refactor / tidy but not today.